src/db/book.rs

```diff
@@ -20,6 +22,7 @@ pub struct Book {
     pub publisher: String,
     pub year: i64,
     pub costs: f64,
+    #[meta(into = Option<String>)]
```

Comment:
Wouldn't something like `#[meta(default = String::new())]` be much easier to work with?
So that `note` is set to "" if it is missing on the typescript side...

Reply:
I cannot really understand what you mean because the only thing that `into` does is replacing the type of the field below for the conversion, resulting here in `string | null`. That's done because the field gets skipped by `serde` on certain conditions.

Reply:
I get it now, you are talking about defaults for these undefined or "missing" fields. I generally don't like the approach of making core logic, which is part of the backend, be handled by the frontend again. Also, the "syntax" of a default field would be in ts code, like `#[meta(default = "")]` or `#[meta(default = false)]` for a boolean.

Reply:
I've now added a better way of making fields optional, but that doesn't resolve the issue of defaults.

Reply:
Yes, I understand the complexity issue. So maybe we do this in the future, but its too much for now.

(However, performance is probably not an issue. Iterating over 200 objects is orders of magnitude faster than the whole network request)